### PR TITLE
feat(polars): Provide polars model schema

### PR DESCRIPTION
**Pull request by jim-ngoo**

### CODE REVIEW

File: src/fastdataframe/polars/model.py [hunk 1]
```diff
@@ -76,5 +84,33 @@ def validate_schema(
...
+        Returns:
+            dict[str, ValidationError]: A dictionary where keys are column names and values are ValidationError
+            instances indicating columns that contain null values.
+        """
+        errors = {}
+        null_count = frame.null_count()
+        if isinstance(frame, pl.LazyFrame):
+            null_count = null_count.collect()
+        for field_name in required_fields:
+            if field_name in frame.columns and null_count[field_name].item() > 0:
```
[the remark this answers is not in the document]
Comment (davzucky) [reply src/fastdataframe/polars/model.py:110]: As well, rather than counting all the null values for the dataframe, you could check only the columns which are nonnullable with `polars.Series.has_nulls` https://docs.pola.rs/api/python/stable/reference/series/api/polars.Series.has_nulls.html

**Comment by jim-ngoo** [comment]
## 🛠️ Kit AI Code Review

Error during enhanced LLM analysis: 'NoneType' object is not subscriptable

---
*Generated by [cased kit](https://github.com/cased/kit) v1.3.1 • Mode: kit • Model: gpt-4o-mini*

**Comment by jim-ngoo** [comment]
## 🛠️ Kit AI Code Review

## Priority Issues

### 1. Missing Type Annotations
- **File**: [src/fastdataframe/polars/model.py:34](https://github.com/davzucky/fastdataframe/blob/3561a712ff4e5ae92863e4300edb8a7a08f3b91b/src/fastdataframe/polars/model.py#L34)
- **Priority**: Medium
- **Issue**: The function `_python_type_to_polars_type` lacks a return type annotation. Adding a return type annotation improves code readability and helps with static type checking.

### 2. Potential KeyError in `_python_type_to_polars_type`
- **File**: [src/fastdataframe/polars/model.py:49](https://github.com/davzucky/fastdataframe/blob/3561a712ff4e5ae92863e4300edb8a7a08f3b91b/src/fastdataframe/polars/model.py#L49)
- **Priority**: Medium
- **Issue**: The function `_python_type_to_polars_type` uses `PYTHON_TO_POLARS_TYPE_MAP.get(py_type, pl.Utf8)`, which defaults to `pl.Utf8` if `py_type` is not found. This could lead to unexpected behavior if the type is not intended to be a string. Consider raising an exception or logging a warning when an unknown type is encountered.

### 3. Lack of Tests for Edge Cases
- **File**: [tests/polars/test_validation.py](https://github.com/davzucky/fastdataframe/blob/3561a712ff4e5ae92863e4300edb8a7a08f3b91b/tests/polars/test_validation.py)
- **Priority**: Medium
- **Issue**: While the tests cover various scenarios, there are no tests for cases where the input DataFrame has mixed types or unsupported types. Adding tests for these edge cases would ensure robustness.

### 4. Inconsistent Naming Conventions
- **File**: [tests/test_models.py:4](https://github.com/davzucky/fastdataframe/blob/3561a712ff4e5ae92863e4300edb8a7a08f3b91b/tests/test_models.py#L4)
- **Priority**: Low
- **Issue**: The field `is_active` is defined with an alias `is active` using `Field(alias=AliasChoices("is_active", "is active"))`. This could lead to confusion. Consider using a consistent naming convention for field names to avoid potential issues with data serialization/deserialization.

## Summary
This PR introduces a new feature to provide a Polars model schema, enhancing the existing functionality of the `PolarsFastDataframeModel`. Key changes include the addition of a mapping from Python types to Polars types, the implementation of a method to generate a Polars schema based on model fields, and the introduction of validation methods for required columns.

## Recommendations
1. **Add Type Annotations**: Ensure all functions, especially `_python_type_to_polars_type`, have complete type annotations for better clarity and type checking.
2. **Handle Unknown Types**: Implement error handling in `_python_type_to_polars_type` to manage unsupported types more gracefully.
3. **Expand Test Coverage**: Include tests for edge cases, such as mixed or unsupported types in the DataFrame, to ensure the robustness of the new functionality.
4. **Standardize Naming Conventions**: Review field names for consistency to prevent confusion during data handling.

By addressing these issues, the code quality and maintainability of the project will be significantly improved.

---
*Generated by [cased kit](https://github.com/cased/kit) v1.3.1 • Mode: kit • Model: gpt-4o-mini*
